### PR TITLE
[Instrumentation.Process] Exposed required metric `process.cpu.count` for computing CPU utilization in the backend.

### PR DESCRIPTION
**Pull request by Yun-Ting**
Based on spec discussion: https://github.com/open-telemetry/opentelemetry-specification/pull/2392#discussion_r832488855
Related to: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/972.

## Changes

Exposed required metric `process.processors` for computing CPU utilization in the backend.

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by Kielek** [comment]
Can we open discussion on spec side once more time?
I would like to avoid some custom metric in the repository created only for one language and then potential subject to change.

**Review by CodeBlanch — APPROVED** [review]
`process.processors` is kind of tongue twister 🤣 `process.cpu.count` is much nicer IMO. Probably `process.threads` -> `process.thread.count` would also be good.

File: src/OpenTelemetry.Instrumentation.Process/README.md [hunk 1]
```diff
@@ -92,6 +92,20 @@ Gets the user processor time for this process.
 * [Process.PrivilegedProcessorTime](https://learn.microsoft.com/dotnet/api/system.diagnostics.process.privilegedprocessortime):
 Gets the privileged processor time for this process.
 
+### process.processors
+
+Process threads count.
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Instrumentation.Process/README.md:97]: `Process threads count.` -> `Process processor count.`

Might be nice to also do like...

`Process processor (CPU) count.`

`Gets the number of processors (CPUs) available to the current process.`

**Comment by Yun-Ting** [comment]
> Can we open discussion on spec side once more time? The minimal effort - decide the name in the spec side. I would like to avoid some custom metric in the repository created only for one language and then potential subject to change.

I've created an issue on the spec side here:
https://github.com/open-telemetry/opentelemetry-specification/issues/3200

**Review by reyang — APPROVED** [review]
LGTM, the spec dependency is not a blocker IMHO given the component and the spec are both preview/experimental.

File: src/OpenTelemetry.Instrumentation.Process/README.md (hunk 1 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.Process/README.md]: updated

File: src/OpenTelemetry.Instrumentation.Process/README.md [hunk 2]
```diff
@@ -92,6 +92,20 @@ Gets the user processor time for this process.
 * [Process.PrivilegedProcessorTime](https://learn.microsoft.com/dotnet/api/system.diagnostics.process.privilegedprocessortime):
 Gets the privileged processor time for this process.
 
+### process.cpu.count
+
+Process processors (CPU cores) count.
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Process/README.md:97]:
```suggestion
The number of processors (CPU cores) available to the current process.
```

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Process/README.md [hunk 3]
```diff
@@ -92,6 +92,20 @@ Gets the user processor time for this process.
...
+
+| Units         | Instrument Type         | Value Type |
+|---------------|-------------------------|------------|
+| `{processors}`| ObservableUpDownCounter | `Int32`    |
+
+The API used to retrieve the value is:
+
+* [Process.processors](https://learn.microsoft.com/dotnet/api/system.environment.processorcount):
+Gets the number of processors available
+to the current process.
```
Comment (reyang) [inline src/OpenTelemetry.Instrumentation.Process/README.md:107]:
```suggestion
The API used to retrieve the value is [System.Environment.ProcessorCount](https://learn.microsoft.com/dotnet/api/system.environment.processorcount).
```

File: src/OpenTelemetry.Instrumentation.Process/README.md [hunk 4]
```diff
@@ -102,6 +102,10 @@ The number of processors (CPU cores) available to the current process.
 
 The API used to retrieve the value is [System.Environment.ProcessorCount](https://learn.microsoft.com/dotnet/api/system.environment.processorcount).
 
+Note:
```
Comment (reyang) [inline src/OpenTelemetry.Instrumentation.Process/README.md:105]: https://github.com/open-telemetry/opentelemetry-dotnet/commit/074c0196f165a6c8df97496363c91784492d92e6

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Process/README.md (hunk 4 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.Process/README.md]: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1005

**Review by Kielek — APPROVED** [review]

**Pull request merged**